### PR TITLE
Blog Post: Authenticate Flask API Using JWT

### PR DESCRIPTION
**Pull request by raghunath-r-a**
#### Please note that in case of below checklist not updated accordingly, the maintainers have the right to immediately close your pull request.

- [x] You created a branch from `dev` branch.
- [x] You PR is raised on `dev` branch and not on`master`.
- [x] Blog follows [Blog Guidelines](https://github.com/LoginRadius/engineering-portal/blob/master/GUIDELINES.md).
- [X] You have read the [Contribution Guidelines](https://github.com/LoginRadius/engineering-portal/blob/master/CONTRIBUTING.md) before creating this PR.
- [X] Checked for any duplicate blog entry.
- [X] Checked for all grammatical errors in the blog.
- [X] You have not published the blog content anywhere else on the internet on platforms like medium, dev.to etc.
- [X] The content is not copy-pasted from anywhere else on the Internet.

Code samples PR: https://github.com/LoginRadius/engineering-blog-samples/pull/126

### CODE REVIEW

**Review by Akash-Patil — DISMISSED** [review]
Hi @Babatunde13 ,

We apologise for the delay.

I have reviewed the blog and requested a few changes. Kindly look into it from your end. 
We will review the code base for the blog once the correct URLs have been updated for the sample code.

Regards, 
@Akash-Patil

File: content/guest-post/how-to-authenticate-flask-api-using-jwt/index.md [hunk 1]
```diff
@@ -0,0 +1,754 @@
...
+
+In this tutorial, you'll work on authentication middleware for an existing API built with Flask and PyMongo. The API is a book library API using which users can create books and upload cover images for the books and relevant data. PyMongo is used to connect to the mongo database. You'll use the PyJWT library to generate and verify JWT tokens. 
+
+To get started, clone the repository and set up the application by running the following commands:
+
+```bash 
+git clone https://github.com/Babatunde13/login-radius-flask.git # Clone the repository
+cd loginRadius-flask-auth # change directory
+git checkout start # check out the start branch
+virtualenv venv # create virtual environment
```
Comment (raghunath-r-a) [inline content/guest-post/how-to-authenticate-flask-api-using-jwt/index.md:56]: @Babatunde13 I guess, you need to update when the reader should use `virtualenv` (Python 2) or `venv` (Python 3) based on https://packaging.python.org/en/latest/guides/installing-using-pip-and-virtual-environments/

File: content/guest-post/how-to-authenticate-flask-api-using-jwt/index.md [hunk 2]
```diff
@@ -0,0 +1,754 @@
...
+source venv/bin/activate # activate virtual environment
```
Comment (raghunath-r-a) [inline content/guest-post/how-to-authenticate-flask-api-using-jwt/index.md:57]: @Babatunde13 This particular command is not working in Windows

Comment (Babatunde13) [reply content/guest-post/how-to-authenticate-flask-api-using-jwt/index.md:57]: ooh yes, thanks for the catch!. Actually, I don't think I need to demonstrate how to create and activate virtual environment as that isn't in the scope of the article.

Comment (raghunath-r-a) [reply content/guest-post/how-to-authenticate-flask-api-using-jwt/index.md:57]: I can understand that. However, it makes it easier for the readers.

Comment (Babatunde13) [reply content/guest-post/how-to-authenticate-flask-api-using-jwt/index.md]: Yes, thanks! I added the commands for windows users also

**Comment by raghunath-r-a** [comment]
Hi @Babatunde13 

Thanks a lot for your perseverance throughout this process!

Our technical reviewer confirmed that everything is working fine now -- and suggested for you to include proper schema format for validating APIs, in the blog post.

Please do this change and then we'll go ahead and publish the blog post. 

Once again, thank you!
@raghunath-r-a

**Comment by Babatunde13** [comment]
Hi @raghunath-r-a, I did use a standard schema for the response (updated it yesterday). the format is 
```
{
    data: something or null,
    error: detailed error object
    message: message to be displayed on the client
}
```

**Review by mohammed786 — APPROVED** [review]

**Comment by raghunath-r-a** [comment]
Hi @Babatunde13 We'll publish your blog soon. Please update your GitHub profile picture to something more relevant, like your own picture or a default gravatar on GitHub. Your GitHub profile picture will be shown in your author bio in the [Async Blog](https://www.loginradius.com/blog/async/).

**Comment by Babatunde13** [comment]
Hi @raghunath-r-a , thanks!
I've updated my profile picture to an illustration of me, I guess that should do. Let me know if there is any other thing you need.

**Pull request merged**